src/pages/api/outboundMatching.ts

```diff
@@ -38,37 +22,31 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
             throw new Error('No service or userfarestages info found');
         }
 
+        const service: BasicService = JSON.parse(req.body.service);
         const userFareStages: UserFareStages = JSON.parse(req.body.userfarestages);
-
         const matchingFareZones = getMatchingFareZonesFromForm(req);
 
         // Deleting these keys from the object in order to faciliate looping through the fare stage values in the body
         delete req.body.service;
         delete req.body.userfarestages;
 
         if (isFareStageUnassigned(userFareStages, matchingFareZones) && matchingFareZones !== {}) {
-            const selectedStagesList: {}[] = getSelectedStages(req);
-
-            const outbound = { error: true, selectedFareStages: selectedStagesList };
-
-            setCookieOnResponseObject(MATCHING_COOKIE, JSON.stringify({ outbound }), req, res);
+            const selectedStagesList: string[] = getSelectedStages(req);
+            updateSessionAttribute(req, MATCHING_ATTRIBUTE, { error: true, selectedFareStages: selectedStagesList });
             redirectTo(res, '/outboundMatching');
             return;
         }
-
-        const uuid = getUuidFromCookie(req, res);
-
         const formatMatchingFareZones = getFareZones(userFareStages, matchingFareZones);
 
         const matchedFareZones: MatchingFareZones = {};
         formatMatchingFareZones.forEach(fareStage => {
             matchedFareZones[fareStage.name] = fareStage;
         });
-
-        await putOutboundMatchingDataInS3(matchedFareZones, uuid);
-
-        setCookieOnResponseObject(MATCHING_COOKIE, JSON.stringify({ outbound: { error: false } }), req, res);
-
+        updateSessionAttribute(req, MATCHING_ATTRIBUTE, {
+            service,
+            userFareStages,
+            matchingFareZones: matchedFareZones,
+        });
```

Comment:
We should create a variable with this information prior to this function call. That way we can assert a type and check that we are formatting the data correctly when we call `updateSessionAttribute`

Reply:
Created a variable and a type (MatchingValues).